src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -391,6 +400,115 @@ public void createMissingClass(UnsolvedClass missedClass) {
     }
   }
 
+  /**
+   * This method capitalizes a string. For example, "hello" will become "Hello".
+   *
+   * @param string the string to be capitalized
+   * @return the capitalized version of the string
+   */
+  public static String toCapital(String string) {
+    return string.substring(0, 1).toUpperCase() + string.substring(1);
+  }
+
+  /**
+   * This method checks if a string is capitalized
+   *
+   * @param string the string to be checked
+   * @return true if the string is capitalized
+   */
+  public static boolean isCapital(String string) {
+    String capitalizedString = toCapital(string);
+    return string.equals(capitalizedString);
+  }
+
+  /**
+   * This method checks if a string is a class path.
+   *
+   * @param potentialClassPath the string to be checked
+   * @return true if the string is a class path
+   */
+  public static boolean isAClassPath(String potentialClassPath) {
+    String[] elements = potentialClassPath.split(".");
+    int elementsCount = elements.length;
+    return elementsCount > 1 && isCapital(elements[elementsCount - 1]);
+  }
+
+  /**
+   * Given a class path, this method will create a simple instance of UnsolvedClass for that class
+   * path
+   *
+   * @param classPath the class path
+   * @return the corresponding instance of UnsolvedClass
+   */
+  public static UnsolvedClass getSimpleSyntheticClassFromClassPath(String classPath) {
+    if (!isAClassPath(classPath)) {
```

Comment:
Rather than `class path` here to describe this, I'd be more comfortable with this code if it was using one of the Signature Checker annotations to describe what this is. I _think_ your definition of `class path` is equivalent to a fully-qualified name (but I'm not 100% sure, which is why I'd like it if we use an annotation, which is definitely unambiguous).